common/src/main/java/us/myles/ViaVersion/protocols/protocol1_9to1_8/packets/InventoryPackets.java

```diff
@@ -145,8 +164,28 @@ public void registerMap() {
                     @Override
                     public void handle(PacketWrapper wrapper) throws Exception {
                         Item[] stacks = wrapper.get(Type.ITEM_ARRAY, 0);
-                        for (Item stack : stacks)
+                        int inventoryId = wrapper.get(Type.UNSIGNED_BYTE, 0);
+
+                        InventoryTracker inventoryTracker = wrapper.user().get(InventoryTracker.class);
+                        EntityTracker1_9 entityTracker = wrapper.user().get(EntityTracker1_9.class);
+
+                        boolean showShieldWhenSwordInHand = Via.getConfig().isShowShieldWhenSwordInHand()
+                                && Via.getConfig().isShieldBlocking();
+
+                        for (short i = 0; i < stacks.length; i++) {
+                            Item stack = stacks[i];
+
+                            // Check if it is the inventory of the player
+                            if (showShieldWhenSwordInHand && inventoryId == 0) {
+                                // Store items in slots
+                                inventoryTracker.getSlotToItemIdMap().put(i, stack == null ? 0 : stack.getIdentifier());
+
+                                // Sync shield item in offhand with main hand
+                                entityTracker.syncShieldWithSword();
```

Comment:
No need to call syncShieldWithSword with each single item transaction, move this outside of the for loop